text/0171-cloudfront-redesign.md

```diff
@@ -0,0 +1,495 @@
+  origin: cloudfront.Origin.fromBucket(myBucket),
+  certificate: myCertificate,
+});
+```
+
+Note that in the above example the aliases are inferred from the certificate and do not need to be explicitly provided.
+
+## Behaviors
+
+Each distribution has a default behavior which applies to all requests to that distribution; additional behaviors may be specified for a
+given URL path pattern. Behaviors allow routing with multiple origins, controlling which HTTP methods to support, whether to require users to
+use HTTPS, and what query strings or cookies to forward to your origin, among others.
+
+The properties of the default behavior can be adjusted as part of the distribution creation. The following example shows configuring the HTTP
+methods and viewer protocol policy of the cache.
+
+```ts
+const myWebDistribution = new cloudfront.Distribution(this, 'myDist', {
+  origin: cloudfront.Origin.fromLoadBalancerV2(myLoadBalancer, {
+    allowedMethods: AllowedMethods.ALL,
+    viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+  }),
+});
+```
+
+Additional cache behaviors can be specified at creation, or added to the origin(s) after the initial creation. These additional cache behaviors enable
+customization for a specific set of resources based on a URL path pattern. For example, we can add a behavior to `myWebDistribution` to override the
+default time-to-live (TTL) for all of the images.
+
+```ts
+myWebDistribution.origin.addBehavior('/images/*.jpg', {
+  viewerProtocolPolicy: ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+  defaultTtl: cdk.Duration.days(7),
+});
+```
+
+## Multiple Origins
+
+A distribution may have multiple origins in addition to the default origin; each additional origin must have (at least) one behavior to route requests
+to that origin. A common pattern might be to serve all static assets from an S3 bucket, but all dynamic content served from a web server. The
+following example shows how such a setup might be created:
+
+```ts
+const myWebsiteBucket = new s3.Bucket(...);
+const myMultiOriginDistribution = new cloudfront.Distribution(this, 'myDist', {
+  origin: cloudfront.Origin.fromWebsiteBucket(myBucket),
+  additionalOrigins: [
+    cloudfront.Origin.fromLoadBalancerV2(myLoadBalancer, {
+      pathPattern: '/api/*',
+      allowedMethods: AllowedMethods.ALL,
+      forwardQueryString: true,
+    }),
+  ],
+});
+```
+
+## Origin Groups
+
+You can specify an origin group for your CloudFront origin if, for example, you want to configure origin failover for scenarios when you need high
+availability. Use origin failover to designate a primary origin for CloudFront plus a second origin that CloudFront automatically switches to when the
+primary origin returns specific HTTP status code failure responses. An origin group can be created and specified as the primary (or additional) origin
+for the distribution.
+
+```ts
+const myOriginGroup = cloudfront.Origin.fromOriginGroup({
```

Comment:
Also, I think `Origin.fromOriginGroup()` is stretching the definition of `from`. It's not like an "origin group" already exists and you're turning it into an origin here.

It feels much more like you're making a new Origin Group, in which case I feel `from` is wrong. 

OTOH, it is kinda discoverable and we generally use `from` for factory functions. Ttssschhh naming is hard!